### PR TITLE
sot-67-investigate-poseidon

### PR DESCRIPTION
**Pull request by Tranduy1dol**
# Check List

- [x] Did you check your grammar / reader proof your document using ChatGPT?
- [x] Don't forget to squash commits into meaningful chunks before merging
- [x] After merging this, please update your task status!
- [x] Did you verify that your math formulas are rendered correctly?

### CODE REVIEW

**Comment by sonntuet1997** [comment]
Fix your commit message format

**Review by bjergsen243 — APPROVED** [review]

File: terms/poseidon_hash_function.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+# Poseidon Hash Function
+
+Reference: [Poseidon Hash Function](https://eprint.iacr.org/2019/458)
+
+## Motivation
+
+The $\mathrm{POSEIDON}$ hash function family is designed based on the $\mathrm{HADES}$ strategy and aims to improve the calculation
+speed and security level of the R1CS circuit and AET circuit.
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:8]: link R1CS to our document

File: terms/poseidon_hash_function.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+A sponge construction builds upon an internal permutation, which is defined by two parameters, the rate (or arity in the context of
+tree hashing) $r$ and the capacity (or inner part) $c$. The rate determines the throughput, the capacity is crucial for the security
+level. This means when fixing the size of the internal permutation to $N$ bits, a tradeoff between throughput and security has to be
+made.
+
+Here is an example, where the construction is used to compute the hash output $h_1\ \parallel\ h_2$ of the block message
+$m_1\ \parallel\ m_2\ \parallel\ m_3\ \parallel\ m_4$,
+where $m_i$ and $h_i$ are $r$-bit values. The initial state $I$ contains all zeros.
+
+![sponge_example.png](../terms/attachments/sponge_example.png)
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:37]: choose other image to show the rate and capacity part

File: terms/poseidon_hash_function.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+#### $\mathrm{POSEIDON}^\pi$ Sponges
+
+---
+
+We provide several $\mathrm{POSEIDON}$ instances for different use cases, but they all use the sponge construction in the same way:
+
+1. Depending on the use case, determine the capacity element value and the input padding if needed.
+2. Split the obtained input into chunks of size $r$.
+3. Apply the permutation $\mathrm{POSEIDON}^\pi$ to the capacity element and the first chunks.
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:55]:
```suggestion
3. Apply the permutation $\mathrm{POSEIDON}^\pi$ to the capacity element and the first chunk.
```

File: terms/poseidon_hash_function.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+exhibit non-generic properties up to $2^M$ queries, where $M$ is the desired security level. For this, we set the capacity to $2M$ and
+denote by $\mathrm{POSEIDON}-M$ a hash function that provides $M$ bits of security against collision and preimage attacks.
+
+#### $\mathrm{POSEIDON}^\pi$ Sponges
+
+---
+
+We provide several $\mathrm{POSEIDON}$ instances for different use cases, but they all use the sponge construction in the same way:
+
+1. Depending on the use case, determine the capacity element value and the input padding if needed.
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:53]:
```suggestion
1. Depending on the use case, determine the value of the capacity element and apply input padding if needed.
```

File: terms/poseidon_hash_function.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+Each round function consists of 3 parts:
+
+1. $AddRoundConstants$, denoted by $ARC(\cdot)$.
+2. $SubWords$, denoted by $\mathrm{S-box}(\cdot)$ or by $SB(\cdot)$.
+3. $MixLayer$, denoted by $M({\cdot})$.
+
+The $MixLayer$ operation is the linear layer, and it consists of multiplying the state a $t\times t$ 
+[MDS matrix](https://en.wikipedia.org/wiki/MDS_matrix), to apply the wide
+trail strategy.
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:94]: what is "the wide trail strategy"?

File: terms/poseidon_hash_function.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+---
+
+Each round function consists of 3 parts:
+
+1. $AddRoundConstants$, denoted by $ARC(\cdot)$.
+2. $SubWords$, denoted by $\mathrm{S-box}(\cdot)$ or by $SB(\cdot)$.
+3. $MixLayer$, denoted by $M({\cdot})$.
+
+The $MixLayer$ operation is the linear layer, and it consists of multiplying the state a $t\times t$
```
Comment (bjergsen243) [inline terms/poseidon_hash_function.md:92]:
```suggestion
The $MixLayer$ operation is the linear layer, and it consists of multiplying the state by a $t\times t$ 
```

**Review by bjergsen243 — APPROVED** [review]

**Review by zk-steve — APPROVED** [review]

**Pull request merged**